satpy/readers/ami_l1b.py

```diff
@@ -75,44 +75,35 @@ def end_time(self):
 
     def get_area_def(self, dsid):
         """Get area definition for this file."""
-        a = self.nc.attrs['earth_equatorial_radius']
-        b = self.nc.attrs['earth_polar_radius']
-        h = self.nc.attrs['nominal_satellite_height'] - a
-        lon_0 = self.nc.attrs['sub_longitude'] * 180 / np.pi  # it's in radians?
-        cols = self.nc.attrs['number_of_columns']
-        rows = self.nc.attrs['number_of_lines']
+        pdict = {}
+        pdict['a'] = self.nc.attrs['earth_equatorial_radius']
+        pdict['b'] = self.nc.attrs['earth_polar_radius']
+        pdict['h'] = self.nc.attrs['nominal_satellite_height'] - pdict['a']
+        pdict['ssp_lon'] = self.nc.attrs['sub_longitude'] * 180 / np.pi  # it's in radians?
+        pdict['ncols'] = self.nc.attrs['number_of_columns']
+        pdict['nlines'] = self.nc.attrs['number_of_lines']
         obs_mode = self.nc.attrs['observation_mode']
         resolution = self.nc.attrs['channel_spatial_resolution']
 
-        cfac = self.nc.attrs['cfac']
-        coff = self.nc.attrs['coff']
-        lfac = self.nc.attrs['lfac']
-        loff = self.nc.attrs['loff']
+        pdict['cfac'] = self.nc.attrs['cfac']
+        pdict['coff'] = self.nc.attrs['coff']
+        pdict['lfac'] = self.nc.attrs['lfac']
+        pdict['loff'] = self.nc.attrs['loff']
+
+        # AMI grid appears offset, we can not use the standard get_area_extent
         bit_shift = 2**16
-        area_extent = (
-            h * np.deg2rad((0 - coff - 0.5) * bit_shift / cfac),
-            -h * np.deg2rad((0 - loff - 0.5) * bit_shift / lfac),
-            h * np.deg2rad((cols - coff + 0.5) * bit_shift / cfac),
-            -h * np.deg2rad((rows - loff + 0.5) * bit_shift / lfac),
-        )
-
-        proj_dict = {
-            'proj': 'geos',
-            'lon_0': float(lon_0),
-            'a': float(a),
-            'b': float(b),
-            'h': h,
-            'units': 'm'
-        }
+        ll_x = (0 - pdict['coff'] - 0.5) * bit_shift / pdict['cfac']
+        ll_y = -(0 - pdict['loff'] - 0.5) * bit_shift / pdict['lfac']
+        ur_x = (pdict['ncols'] - pdict['coff'] + 0.5) * bit_shift / pdict['cfac']
+        ur_y = -(pdict['nlines'] - pdict['loff'] + 0.5) * bit_shift / pdict['lfac']
+
+        area_extent = make_ext(ll_x, ur_x, ll_y, ur_y, pdict['h'])
```

Comment:
What is it that makes this not factorizable to `get_area_extent` ?

Reply:
As mentioned on slack, AMI produces an incorrect area definition if the new methods are used. This is likely due to AMI using centre-of-pixel rather than corner-of-pixel or vice-versa.